.env.sample

```diff
@@ -5,6 +5,7 @@ PORT=
 NODE_ENV=development
 TZ=Europe/London
 LOG_LEVEL=info
+REDACT_SENSITIVE_DATA_IN_LOGS=false
```

Comment:
@abhi-markan can you please add this environment variable to GitHub please? We want it to be `true` in all our deployed environments please. (It's `false` here because for development it's more useful to see the full log messages and there's much less risk of logging the data)

Reply:
I wonder if it's worth using the `LOG_LEVEL` instead to determine whether the data is redacted or not. Meaning, if:
1. `LOG_LEVEL` = info, then it's redacted (because it only returns basic information about the payload). 
2. `LOG_LEVEL` = debug, then it returns everything (because in debug mode you'd want to know where issues occur)

What do you think?

Reply:
From my experience sensitive data is not used for most of issue investigations. 
It is useful to have higher `LOG_LEVEL` to understand issues with client, API or end system, so not sure what LOG_LEVEL we plan to use in Production.

Voting for separate environment variables.

Reply:
I agree with Audrius' vote here: we've managed to leave most of the debug logging data unredacted so I expect that we can leave the redacting on 99% of the time (regardless of the log level we choose) and only turn it off on the rare occasion that we need the redacted data

Reply:
ok cool. can we rename the variable into something like `REDACT_LOGS`? This to avoid attracting too much attention with the env variable name :)

Reply:
p.s. is there a time when we want to see the credentials though? i.e. for localhost we can use `console.info` if we want to check for anything.

Reply:
I've renamed the config to `REDACT_LOGS` / `redactLogs`

> for localhost we can use console.info if we want to check for anything

The environment variable is more for the deployed application than it is for localhost. If we need to debug an issue between DTFS -> APIM -> TFS in staging or production then we might need to view things like the client's headers which we're redacting (e.g. to understand why authentication is failing). I think changing only the environment variables + redeploying is quicker and easier than creating a new version of the code + redeploying - although that is up for debate :)  

The other thing is that not all of these values are easy to get hold of to pass into `console.info`. For example, to log the client's headers we'd need to get hold of the Express request object which none of our controllers do currently. So we'd need to check if we could put `console.info` in one of our interceptors that use the request object instead, or change the controller code so that NestJs gives the controller the Express request object for us to log. It's not impossible, but a bit more work to do it (without breaking any tests) than you'd think.

I think if it's a use case that we expect to use then it's better to support it properly and make it easier to sure, but I'm happy to be told otherwise if people disagree :)

Reply:
makes sense. thanks!